Updated the login and search test cases as per modified framework

amit-k-gupta2 opened the pull request.

Commit:
Updated the login and search test cases as per modified framework

dil-sjain commented on cypress-journey/.env.example:
remove this file from commit

dil-sjain commented on cypress-journey/.idea/.gitignore:
remove this file from commit

dil-sjain commented on cypress-journey/cypress/e2e/loginTest.cy.js:
```diff
@@ -0,0 +1,50 @@
+// I intentionally add hard waits to mimic human interaction speed.
+// This helps avoid triggering Amazon's bot protection (e.g., Captcha challenges).
+describe("Amazon Login Tests", () => {
+  beforeEach(() => {
```
why can't we use before hook here?

amit-k-gupta2 replied:
It was creating issue.

dil-sjain commented on cypress-journey/cypress/e2e/loginTest.cy.js:
```diff
@@ -0,0 +1,50 @@
+import loginPage from "../support/pageObjects/loginPage";
+import homePage from "../support/pageObjects/homePage";
+
+// I intentionally add hard waits to mimic human interaction speed.
```
the primary goal of automation is to increase speed and efficiency, remove hard waits

amit-k-gupta2 replied:
At all the places , intentionally added hard wait to simulate human behavior, otherwise Amazon will block account considering it a bot . One of my account was blocked by Amazon and now they are not allowing to sign-in even after taking support from Customer Service. 
Similarly in new PR I will add hard wait intentionally.

dil-sjain commented on cypress-journey/cypress/e2e/loginTest.cy.js:
```diff
@@ -0,0 +1,50 @@
+    loginPage.enterPassword(Cypress.env("password"));
+    cy.customWait(2);
+    loginPage.clickOnLoginButton();
+    loginPage.validateErrorMessage("There was a problem");
```
keep all error/validation strings in fixtures

amit-k-gupta2 replied:
Noted and updated

dil-sjain commented on cypress-journey/cypress/e2e/loginTest.cy.js:
```diff
@@ -0,0 +1,50 @@
+    cy.customWait(2);
+    loginPage.clickOnContinueButton();
+    cy.customWait(2);
+    loginPage.enterPassword(Cypress.env("password"));
```
these seems to be valid password not invalid

amit-k-gupta2 replied:
corrected

dil-sjain commented on cypress-journey/cypress/fixtures/searchData.json:
```diff
@@ -0,0 +1,5 @@
+{
+  "validSearches": ["iPhone 13", "Samsung Galaxy", "Laptop"],
+  "invalidSearches": ["asdkjasdkjas", "#######"],
+  "emptySearch": "{backspace}"
```
this can be used by simply using type('{backspace}'), instead of defining here

amit-k-gupta2 replied:
Since here test cases are data driven so used this way

dil-sjain commented on cypress-journey/cypress/support/commands.js:
```diff
@@ -1,12 +1,17 @@
+  loginPage.clickOnLoginButton();
+});
+
+Cypress.Commands.add("customWait", (seconds) => {
```
create a util function for this not custom command

amit-k-gupta2 replied:
Tried but was facing issue so restored to this approach.

dil-sjain commented on cypress-journey/cypress/support/e2e.js:
remove this from commit

amit-k-gupta2 replied:
This have command import statement without which error is occurring so can remove it.

dil-sjain commented on cypress-journey/package-lock.json:
remove this from commit

amit-k-gupta2 replied:
Will remove in further commit. While reading about framework , I found that it works like pom.xml of MAVEN so that every one should be in sync so added it . And addition of it was recommended. But as you asked to remove so will remove.

Commit:
Enhance Product Purchase Flow Automation and Implement Session Save + Logout

 Incorporated review feedback and made necessary corrections in the existing framework.
- Added a full end-to-end test case:
  - Search products using keyword.
  - Apply filters and validate the search results.
  - Add selected products to cart.
  - Validate cart contents, delete an item, and proceed to checkout.
  - Perform login and validate delivery address fields.
  - Validate total amount at checkout matches calculated cart total.
  - Empty cart post-validation and perform logout.

- Introduced new session handling test:
  - Save user session information after successful login.
  - Logout from Amazon using API for faster execution.

- Added methods for cart price calculation and validation.
- Saved total cart amount into a JSON file for future reference.
- Improved error handling and robustness by adding custom waits and retries.
- Ensured logout is handled both through UI and API as per scenario needs.

- Verified product search and filtering across multiple datasets.
- Confirmed accurate cart validations.
- Successfully tested session save and API-based logout functionality.

Commit:
Removed unwanted files

Commit:
removed unnecessay files

Commit:
removed unnecessary file

amit-k-gupta2 commented:
[Uploading loginTest.cy.js.zip…]()
[Uploading searchTest.cy.js.zip…]()

amit-k-gupta2 commented:
I think compressed files are not uploaded properly so adding drive link https://drive.google.com/drive/folders/1iq3Xsbv6xeIpcm7nyFjbuFxO7DxKyoXv?usp=drive_link

dil-sjain commented on cypress-journey/.env.example:
remove this file from commit

dil-sjain commented on cypress-journey/.gitignore:
remove this file from commit

dil-sjain commented on cypress-journey/cypress.config.js:
```diff
@@ -1,22 +1,30 @@
-    BASE_URL: process.env.BASE_URL,
+    mobileNum: process.env.MOBILENUM,
+    password: process.env.PASSWORD,
+    invalidPassword: process.env.INVALIDPASSWORD,
```
it would be better if we dont store negative scenarios through env,
Instead, handle such test data dynamically within the test itself or through fixtures.

amit-k-gupta2 replied:
In the beginning, I have used the data file only for login credentials but since we switched to env file so I kept all types of credentials here only.

dil-sjain commented on cypress-journey/cypress/e2e/sample.cy.js:
remove this file from commit

dil-sjain commented on cypress-journey/cypress/e2e/searchTest.cy.js:
```diff
@@ -0,0 +1,91 @@
+    });
+  });
+
+  it("Search a product , add to cart ,proceed to checkout,login , validate delivery address and validate amount", () => {
```
can we break this flow and handle some functions to reduce line of codE?

amit-k-gupta2 replied:
This was e2e scenario , that's why it is lengthy

dil-sjain commented on cypress-journey/cypress/fixtures/example.json:
remove this file from commit
